src/ntfsck.c

```diff
@@ -2374,6 +2374,172 @@ static int ntfsck_check_index_bitmap(ntfs_inode *ni, ntfs_attr *bm_na)
 	return STATUS_OK;
 }
 
+static void ntfsck_validate_index_blocks(ntfs_volume *vol,
+					 ntfs_index_context *ictx)
+{
+	INDEX_ALLOCATION *ia;
+	FILE_NAME_ATTR *ie_fn;
+	INDEX_ENTRY *ie;
+	INDEX_ROOT *ir = ictx->ir;
+	ntfs_inode *ni = ictx->ni, *cni;
+	MFT_REF mref;
+	VCN vcn;
+	u32 ir_size = le32_to_cpu(ir->index.index_length);
+	u32 ib_cnt = 1, i;
+	BOOL ib_corrupted = FALSE;
+	u8 *ir_buf, *ia_buf = NULL, *ibs, *index_end;
+
+	ictx->ia_na = ntfs_attr_open(ni, AT_INDEX_ALLOCATION,
+			ictx->name, ictx->name_len);
+	if (!ictx->ia_na)
+		return;
+
+	ir_buf = malloc(le32_to_cpu(ir->index.index_length));
```

Comment:
if ir_buf is not used before checking if index block is corrupted (before line 2430), I think it's better to move this line after checking (after line 2430)

Reply:
Thanks for your review:), I have already applied this patch to #master. So you can improve it and please send a PR.